src/docker/composeappengine.cc

```diff
@@ -322,6 +324,37 @@ bool ComposeAppEngine::start(const App& app) {
   return result;
 }
 
+bool ComposeAppEngine::areContainersCreated(const App& app) {
+  bool result{false};
+  try {
+    ComposeInfo info((appRoot(app) / ComposeFile).string());
+    std::vector<Json::Value> services = info.getServices();
+    if (services.empty()) {
+      LOG_WARNING << "App: " << app.name << ", no services in docker file!";
+      return false;
```

Comment:
maybe we need defensive code here, but shouldn't this really go into `verify` so that we can reject an app with no services?

Reply:
Yep, this is rather to make "defensive code".
It looks like something that not exactly related to the given PR, although makes sense to be improved. On the other side, our CI should prevent from such cases.
Anyway, I think, I am fine with adding a check for service number to the verification step.

Reply:
I think it makes sense to address it in another PR because I see that we process the target verification failure incorrectly and there is not test for the given case, so all this together makes it quite a big change.

Reply:
k. lets do LOG_ERROR here and address that in another PR.

Reply:
force pushed it.